ray-operator/controllers/ray/rayservice_controller.go

```diff
@@ -556,12 +556,29 @@ func (r *RayServiceReconciler) cleanUpRayClusterInstance(ctx context.Context, ra
 		return err
 	}
 
+	// Determine the ray cluster deletion delay seconds
+	deletionDelay := RayClusterDeletionDelayDuration
+	if rayServiceInstance.Spec.RayClusterDeletionDelaySeconds != nil {
+		if *rayServiceInstance.Spec.RayClusterDeletionDelaySeconds < 0 {
+			logger.Info(
+				"RayClusterDeletionDelaySeconds is set to a negative value; using default deletion delay.",
+				"RayClusterDeletionDelaySeconds", *rayServiceInstance.Spec.RayClusterDeletionDelaySeconds,
+				"deletionDelay", RayClusterDeletionDelayDuration.String(),
+			)
+		} else {
+			deletionDelay = time.Duration(*rayServiceInstance.Spec.RayClusterDeletionDelaySeconds) * time.Second
+			logger.Info("RayClusterDeletionDelaySeconds is set correctly and will be used.",
```

Comment:
I wonder if the logs are too verbose. Will they be printed for every reconciliation before the shutdown is complete?